src/decoderbufs.c

```diff
@@ -269,10 +269,13 @@ static void set_datum_value(Decoderbufs__DatumMessage *datum_msg, Oid typid,
       datum_msg->datum_case = DECODERBUFS__DATUM_MESSAGE__DATUM_DATUM_INT32;
       break;
     case INT8OID:
-    case OIDOID:
       datum_msg->datum_int64 = DatumGetInt64(datum);
       datum_msg->datum_case = DECODERBUFS__DATUM_MESSAGE__DATUM_DATUM_INT64;
       break;
+    case OIDOID:
+      datum_msg->datum_int64 = (Oid) DatumGetUInt64(datum);
```

Comment:
So does `datum_int64` take the full unsigned value range?

Reply:
https://www.postgresql.org/docs/13/datatype-oid.html
> The oid type is currently implemented as an unsigned four-byte integer. Therefore, it is not large enough to provide database-wide uniqueness in large databases, or even in large individual tables.

int64 is 8byte integer so it is more than enough to accomodate unsigned 4bytes int.

Reply:
Ah, I see. I missed that OID is an unsigned int32 not uint64. Thx!